mungegithub/mungers/milestone-maintainer_test.go

```diff
@@ -111,177 +112,380 @@ func TestMilestoneMaintainer(t *testing.T) {
 	server.Close()
 }
 
-func milestoneTestComment(label string, context string, createdAt time.Time) *c.Comment {
-	n := &c.Notification{
-		Name:      milestoneNotifierName,
-		Arguments: label,
-		Context:   context,
-	}
-	return &c.Comment{
-		Body:      stringPtr(n.String()),
-		CreatedAt: &createdAt,
-	}
-}
-
-func milestoneLabelEvents(label string, createdAt time.Time) []*githubapi.IssueEvent {
-	return []*githubapi.IssueEvent{
-		{
-			Event: stringPtr("labeled"),
-			Label: &githubapi.Label{
-				Name: &label,
-			},
-			CreatedAt: &createdAt,
-			Actor: &githubapi.User{
-				Login: stringPtr(milestoneTestBotName),
-			},
-		},
+// TestNewIssueChangeConfig validates the creation of an IssueChange
+// for a given issue state.
+func TestNewIssueChangeConfig(t *testing.T) {
+	const incompleteLabels = `
+_**kind**_: Must specify exactly one of ` + "`kind/bug`, `kind/cleanup` or `kind/feature`." + `
+_**sig owner**_: Must specify at least one label prefixed with ` + "`sig/`." + `
+`
+	const blockerCompleteLabels = `
+<summary>Issue Labels</summary>
+
+- ` + "`sig/foo`: Issue will be escalated to these SIGs if needed." + `
+- ` + "`priority/critical-urgent`: Never automatically move out of a release milestone; continually escalate to contributor and SIG through all available channels." + `
+- ` + "`kind/bug`: Fixes a bug discovered during the current release." + `
+</details>
+`
+
+	const nonBlockerCompleteLabels = `
+<summary>Issue Labels</summary>
+
+- ` + "`sig/foo`: Issue will be escalated to these SIGs if needed." + `
+- ` + "`priority/important-soon`: Escalate to the issue owners and SIG owner; move out of milestone after several unsuccessful escalation attempts." + `
+- ` + "`kind/bug`: Fixes a bug discovered during the current release." + `
+</details>
+`
+
+	munger := &MilestoneMaintainer{
+		botName:             milestoneTestBotName,
+		activeMilestone:     "v1.8",
+		labelGracePeriod:    3 * day,
+		approvalGracePeriod: 7 * day,
+		warningInterval:     day,
+		slushUpdateInterval: 3 * day,
+		freezeDate:          "the time heck freezes over",
 	}
-}
 
-// TestNotificationState validates how the notificationState() method
-// responds to the possible label and notification states of issues in
-// a milestone.
-func TestNotificationState(t *testing.T) {
 	createdNow := time.Now()
-	warningInterval := day
-	createdYesterday := createdNow.Add(-(warningInterval + time.Hour))
-	gracePeriod := 3 * day
-	createdLongAgo := createdNow.Add(-(gracePeriod + time.Hour))
+	// createdPastWarningInterval := createdNow.Add(-(munger.warningInterval + time.Hour))
+	createdPastLabelGracePeriod := createdNow.Add(-(munger.labelGracePeriod + time.Hour))
+	createdPastApprovalGracePeriod := createdNow.Add(-(munger.approvalGracePeriod + time.Hour))
+	createdPastSlushUpdateInterval := createdNow.Add(-(munger.slushUpdateInterval + time.Hour))
+
 	tests := map[string]struct {
```

Comment:
Could you also call `updateMilestoneStateLabel` during these tests and validate that the correct labels are added and removed?

Reply:
This test is intentionally avoiding modification of the object since setup and validation of that is necessarily mock-heavy.  `TestMilestoneMaintainer` does validate that things hang together - comment and label addition are validated there - but comment deletion will require some messy mocking.  Would it be acceptable to rely on manual validation for now?

Reply:
Yeah thats fine, I didn't see that `TestMilestoneMaintainer` was checking the labels on the issue. Any errors related to labeling would be apparent in the dry run too.